In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/dataset/order_data_last_six_month.xlsx - Worksheet.csv
/kaggle/input/dataset/associated_order_item_data_last_six_month.xlsx - Worksheet.csv


In [57]:
df_order = pd.read_csv('/kaggle/input/dataset/order_data_last_six_month.xlsx - Worksheet.csv')
df_associated_order = pd.read_csv('/kaggle/input/dataset/associated_order_item_data_last_six_month.xlsx - Worksheet.csv')

In [58]:
df_order

order_date  order_id     order_number  customer_id  \
0      03/04/2025    136349  SO/25-26/000818         5235   
1      30/01/2025    110393   SO/24-25/90881         7622   
2      12/02/2025    116427   SO/24-25/96511         3827   
3      26/12/2024     96182   SO/24-25/77411         2223   
4      30/12/2024     97858   SO/24-25/78975         6903   
...           ...       ...              ...          ...   
64454  26/12/2024     96152   SO/24-25/77382          948   
64455  26/12/2024     96153   SO/24-25/77392         5822   
64456  25/12/2024     96166   SO/24-25/77390         1655   
64457  26/12/2024     96173   SO/24-25/77401         4268   
64458  26/12/2024     96175   SO/24-25/77400          998   

                              customer_name              poc_name  poc_id  \
0            Anshu General Store Sector 34            Vikas Gupta       6   
1                            bhai di rasoi          Abhay Srimali    7814   
2                Shama chicken corner beta            Yash Pandey      51   
3                 Bikaner Sweets ( Harola )             Swatantra      25   
4                      Lucknow kirana store             Raj Kumar    7039   
...                                     ...                   ...     ...   
64454                            Maa Bindwa  Utkarsh Pratap Singh      17   
64455             Yaduvanshi General Store             Raza Rizvi    5602   
64456  Brijwasi sweet and restaurant  (115)     Ashutosh Tripathi      37   
64457                      Ashutosh Traders           Mohd Yaqoob    4293   
64458                 Shawarna Wala Al Baik         Nirbhay Singh      24   

         amount  discount  net_order_amount  profit order_status  \
0      19125.00   1650.00           17475.0     0.0       CLOSED   
1       9670.48    735.48            8935.0   -56.4       CLOSED   
2      62500.00   5000.00           57500.0  -650.0       CLOSED   
3       4520.00    420.00            4100.0    40.0       CLOSED   
4       1700.00      0.00            1700.0   -88.0       CLOSED   
...         ...       ...               ...     ...          ...   
64454   1650.00    130.00            1520.0   -60.0       CLOSED   
64455   8255.00    935.00            7320.0  -455.0       CLOSED   
64456  20970.00   2140.00           18830.0  -195.0       CLOSED   
64457   3508.00    198.00            3310.0   -30.0       CLOSED   
64458  10690.00    700.00            9990.0    10.0       CLOSED   

      warehouse_name  warehouse_id  
0              Noida             3  
1      Greater NOIDA             6  
2              Noida             3  
3              Noida             3  
4        Gomti Nagar             1  
...              ...           ...  
64454    Gomti Nagar             1  
64455       Telibagh             2  
64456          Noida             3  
64457       Telibagh             2  
64458    Gomti Nagar             1  

[64459 rows x 14 columns]

In [60]:
# Find names linked to multiple unique customer IDs
duplicate_names = df_order.groupby('customer_name')['customer_id'].nunique()
duplicate_names = duplicate_names[duplicate_names > 1]

print("Customers with same name but different IDs:")
print(duplicate_names)

Customers with same name but different IDs:
customer_name
93 Bataliyan CRPF                2
Ahmad Kirana Store               2
Anoop General Store              2
Awasthi General Store            2
Balaji Kirana Store              2
Balaji Sweets                    2
Bangali Sweets                   2
CHAURASIYA GENERAL STORE         2
Champaran Meat House             2
Deepak General Store             2
GUPTA GENERAL STORE              2
Guddu general Store              2
Gupta General Store              3
Gupta General Store              3
Gupta Puri Bhandar               2
Gupta sweet shop                 2
Harsh General Store              2
Jagdeesh General Store           2
Jaiswal General Store            2
Jaiswal Kirana Store             4
Janta Dhaba                      2
Janta General Store              2
M/S SWAD DAIRY AND SWEETS        3
Mahadev Traders                  2
Mahi General Store               2
Mohan Kirana Store               2
Mohit General Store             

In [61]:
# Find names linked to multiple unique customer IDs
duplicate_names = df_order.groupby('poc_name')['poc_id'].nunique()
duplicate_names = duplicate_names[duplicate_names > 1]

print("POC with same name but different IDs:")
print(duplicate_names)


POC with same name but different IDs:
poc_name
Harshit Shukla    2
Vikas Gupta       3
Name: poc_id, dtype: int64


In [4]:
df_associated_order

order_id     order_number  order_item_id                    item_name  \
0         104592   SO/24-25/85358         202967                 Anannya Rice   
1          74112   SO/24-25/55656         148560           Ambey Patent Maida   
2         105873   SO/24-25/86555         205222                    Arhar Dal   
3         112302   SO/24-25/92717         216104                  Kezar Maida   
4          74308   SO/24-25/55887         148902                 Normal Sugar   
...          ...              ...            ...                          ...   
115088    139435  SO/25-26/003737         262494              SAPNA GOLD AATA   
115089    139433  SO/25-26/003738         262491                 Hum Tum Rice   
115090    139431  SO/25-26/003739         262489  Masodha Double Filter Sugar   
115091    139430  SO/25-26/003740         262488                        Matar   
115092    139429  SO/25-26/003741         262487              SAPNA GOLD AATA   

        quantity  invoiced_quantity       mrp  price_per_unit   amount  \
0              2                  2  1732.000          1534.0  3464.00   
1              1                  1  1899.000          1760.0  1899.00   
2             10                  0   173.567           127.0  1735.67   
3              5                  5  1760.000          1652.0  8800.00   
4              1                  1  2190.000          2110.0  2190.00   
...          ...                ...       ...             ...      ...   
115088         2                  0  2010.000          1470.0  4020.00   
115089         5                  0  1390.000          1215.0  6950.00   
115090         4                  0  2430.000          2210.0  9720.00   
115091        10                  0    54.675            45.0   546.75   
115092         1                  0  2010.000          1450.0  2010.00   

        discount_amount  profit  
0                396.00 -208.00  
1                139.00  -10.00  
2                465.67 -377.00  
3                540.00 -140.00  
4                 80.00   40.00  
...                 ...     ...  
115088          1080.00 -780.00  
115089           875.00 -375.00  
115090           880.00 -160.00  
115091            96.75  -56.75  
115092           560.00 -410.00  

[115093 rows x 11 columns]

In [5]:
df_order.columns , df_associated_order.columns

(Index(['order_date', 'order_id', 'order_number', 'customer_id',
        'customer_name', 'poc_name', 'poc_id', 'amount', 'discount',
        'net_order_amount', 'profit', 'order_status', 'warehouse_name',
        'warehouse_id'],
       dtype='object'),
 Index(['order_id', 'order_number', 'order_item_id', 'item_name', 'quantity',
        'invoiced_quantity', 'mrp', 'price_per_unit', 'amount',
        'discount_amount', 'profit'],
       dtype='object'))

In [6]:
df_associated_order.nunique()

order_id              64459
order_number          64459
order_item_id        115093
item_name               607
quantity                117
invoiced_quantity       128
mrp                    1992
price_per_unit         2216
amount                 8264
discount_amount        6429
profit                 4614
dtype: int64

# Preprocessing the Order Dataset

In [7]:
df_order.describe()

order_id   customer_id        poc_id        amount       discount  \
count   64459.000000  64459.000000  64459.000000  6.445900e+04   64459.000000   
mean   103426.837307   4386.190447   2920.965466  1.606811e+04    1350.906929   
std     20749.791913   2391.437029   3046.872786  3.308837e+04    3582.402681   
min     67415.000000      1.000000      2.000000  1.489000e+01       0.000000   
25%     85413.500000   2355.000000     35.000000  2.260000e+03     150.000000   
50%    103379.000000   4541.000000     52.000000  5.113970e+03     375.000000   
75%    121395.500000   6216.000000   5602.000000  1.565000e+04    1200.000000   
max    139441.000000   9367.000000   9555.000000  1.010000e+06  389858.000000   

       net_order_amount        profit  warehouse_id  
count      64459.000000  64459.000000  64459.000000  
mean       14717.198282   -308.391379      2.323043  
std        30249.534820   1454.542967      1.348273  
min           14.890000 -86382.000000      1.000000  
25%         2090.000000   -203.340000      1.000000  
50%         4680.000000    -50.000000      2.000000  
75%        14333.000000     -5.000000      3.000000  
max       950000.000000  84750.000000      6.000000

In [8]:
df_order.nunique()

order_date            190
order_id            64459
order_number        64459
customer_id          4161
customer_name        4093
poc_name               94
poc_id                 97
amount              16440
discount             9525
net_order_amount    13168
profit               6643
order_status            2
warehouse_name          6
warehouse_id            6
dtype: int64

There are total of 64459 rows

Checking the Datatypes

In [9]:
df_order.dtypes

order_date           object
order_id              int64
order_number         object
customer_id           int64
customer_name        object
poc_name             object
poc_id                int64
amount              float64
discount            float64
net_order_amount    float64
profit              float64
order_status         object
warehouse_name       object
warehouse_id          int64
dtype: object

In [10]:
df_order['order_date'] = pd.to_datetime(df_order['order_date'], format="%d/%m/%Y")


Checking for the missing value in the Dataset

In [11]:
df_order.isnull().sum()


order_date          0
order_id            0
order_number        0
customer_id         0
customer_name       0
poc_name            0
poc_id              0
amount              0
discount            0
net_order_amount    0
profit              0
order_status        0
warehouse_name      0
warehouse_id        0
dtype: int64

No missing Values

# # Categorical Labeling the Order Status Column

0: CLOSED

1: OPEN

In [12]:
df_order['order_status'] = df_order['order_status'].str.lower().map({'open': 1, 'closed': 0})


Number of Order Open and Closed

In [13]:
df_order['order_status'].value_counts()

order_status
0    64317
1      142
Name: count, dtype: int64

In [14]:
df_order[['order_date', 'order_id', 'order_number']].head(20)


order_date  order_id     order_number
0  2025-04-03    136349  SO/25-26/000818
1  2025-01-30    110393   SO/24-25/90881
2  2025-02-12    116427   SO/24-25/96511
3  2024-12-26     96182   SO/24-25/77411
4  2024-12-30     97858   SO/24-25/78975
5  2025-01-04     99588   SO/24-25/80618
6  2025-01-08    101333   SO/24-25/82271
7  2025-01-12    103091   SO/24-25/83927
8  2025-01-17    104851   SO/24-25/85612
9  2025-01-21    106675   SO/24-25/87331
10 2025-02-04    112290   SO/24-25/92679
11 2025-02-08    114279   SO/24-25/94534
12 2025-01-25    108509   SO/24-25/89102
13 2025-02-18    119153   SO/24-25/99058
14 2025-03-01    124372  SO/24-25/103987
15 2025-02-18    119154   SO/24-25/99075
16 2024-12-26     96150   SO/24-25/77380
17 2024-12-30     97859   SO/24-25/78979
18 2025-01-04     99589   SO/24-25/80617
19 2025-01-21    106676   SO/24-25/87528

Expanding the order number column

In [15]:
df_order['order_number'].value_counts()

order_number
SO/25-26/000818    1
SO/24-25/52316     1
SO/24-25/53405     1
SO/24-25/53764     1
SO/24-25/54219     1
                  ..
SO/24-25/107725    1
SO/24-25/109993    1
SO/24-25/79974     1
SO/24-25/81575     1
SO/24-25/77400     1
Name: count, Length: 64459, dtype: int64

In [16]:
df_order[['order_type', 'fiscal_year', 'order_sequence_num']] = df_order['order_number'].str.split('/', expand=True)

In [17]:
df_order.nunique()

order_date              190
order_id              64459
order_number          64459
customer_id            4161
customer_name          4093
poc_name                 94
poc_id                   97
amount                16440
discount               9525
net_order_amount      13168
profit                 6643
order_status              2
warehouse_name            6
warehouse_id              6
order_type                1
fiscal_year               2
order_sequence_num    64459
dtype: int64

In [18]:
df_order.dtypes

order_date            datetime64[ns]
order_id                       int64
order_number                  object
customer_id                    int64
customer_name                 object
poc_name                      object
poc_id                         int64
amount                       float64
discount                     float64
net_order_amount             float64
profit                       float64
order_status                   int64
warehouse_name                object
warehouse_id                   int64
order_type                    object
fiscal_year                   object
order_sequence_num            object
dtype: object

Droping unneccessary Columns

In [19]:
df_order.drop(columns = ['order_number', 'customer_name', 'poc_name', 'warehouse_name', 'order_type', 'order_sequence_num'], inplace = True)

Since, there are only 2 types of values in fiscal year, so we're converting it into a categorical data

In [20]:
df_order['fiscal_year'] = df_order['fiscal_year'].astype('category')

In [21]:
df_order

order_date  order_id  customer_id  poc_id    amount  discount  \
0     2025-04-03    136349         5235       6  19125.00   1650.00   
1     2025-01-30    110393         7622    7814   9670.48    735.48   
2     2025-02-12    116427         3827      51  62500.00   5000.00   
3     2024-12-26     96182         2223      25   4520.00    420.00   
4     2024-12-30     97858         6903    7039   1700.00      0.00   
...          ...       ...          ...     ...       ...       ...   
64454 2024-12-26     96152          948      17   1650.00    130.00   
64455 2024-12-26     96153         5822    5602   8255.00    935.00   
64456 2024-12-25     96166         1655      37  20970.00   2140.00   
64457 2024-12-26     96173         4268    4293   3508.00    198.00   
64458 2024-12-26     96175          998      24  10690.00    700.00   

       net_order_amount  profit  order_status  warehouse_id fiscal_year  
0               17475.0     0.0             0             3       25-26  
1                8935.0   -56.4             0             6       24-25  
2               57500.0  -650.0             0             3       24-25  
3                4100.0    40.0             0             3       24-25  
4                1700.0   -88.0             0             1       24-25  
...                 ...     ...           ...           ...         ...  
64454            1520.0   -60.0             0             1       24-25  
64455            7320.0  -455.0             0             2       24-25  
64456           18830.0  -195.0             0             3       24-25  
64457            3310.0   -30.0             0             2       24-25  
64458            9990.0    10.0             0             1       24-25  

[64459 rows x 11 columns]

In [22]:
print("Start date:", df_order['order_date'].min())
print('End Date:', df_order['order_date'].max())

Start date: 2024-09-30 00:00:00
End Date: 2025-04-10 00:00:00


In [23]:
df_order.fiscal_year.value_counts()

fiscal_year
24-25    60913
25-26     3546
Name: count, dtype: int64

In [24]:
from ydata_profiling import ProfileReport  # If using latest version of pandas-profiling

# Generate profile reports
profile_orders = ProfileReport(df_order, title="Orders Data Report", explorative=True)
# Save as HTML
profile_orders.to_file("df_orders_visualization.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]


100%|██████████| 11/11 [00:00<00:00, 23.50it/s][A


Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

# Preprocessing the Item Dataset

In [25]:
df_associated_order

order_id     order_number  order_item_id                    item_name  \
0         104592   SO/24-25/85358         202967                 Anannya Rice   
1          74112   SO/24-25/55656         148560           Ambey Patent Maida   
2         105873   SO/24-25/86555         205222                    Arhar Dal   
3         112302   SO/24-25/92717         216104                  Kezar Maida   
4          74308   SO/24-25/55887         148902                 Normal Sugar   
...          ...              ...            ...                          ...   
115088    139435  SO/25-26/003737         262494              SAPNA GOLD AATA   
115089    139433  SO/25-26/003738         262491                 Hum Tum Rice   
115090    139431  SO/25-26/003739         262489  Masodha Double Filter Sugar   
115091    139430  SO/25-26/003740         262488                        Matar   
115092    139429  SO/25-26/003741         262487              SAPNA GOLD AATA   

        quantity  invoiced_quantity       mrp  price_per_unit   amount  \
0              2                  2  1732.000          1534.0  3464.00   
1              1                  1  1899.000          1760.0  1899.00   
2             10                  0   173.567           127.0  1735.67   
3              5                  5  1760.000          1652.0  8800.00   
4              1                  1  2190.000          2110.0  2190.00   
...          ...                ...       ...             ...      ...   
115088         2                  0  2010.000          1470.0  4020.00   
115089         5                  0  1390.000          1215.0  6950.00   
115090         4                  0  2430.000          2210.0  9720.00   
115091        10                  0    54.675            45.0   546.75   
115092         1                  0  2010.000          1450.0  2010.00   

        discount_amount  profit  
0                396.00 -208.00  
1                139.00  -10.00  
2                465.67 -377.00  
3                540.00 -140.00  
4                 80.00   40.00  
...                 ...     ...  
115088          1080.00 -780.00  
115089           875.00 -375.00  
115090           880.00 -160.00  
115091            96.75  -56.75  
115092           560.00 -410.00  

[115093 rows x 11 columns]

In [26]:
df_associated_order.drop(columns = ['order_number'], inplace = True)

In [27]:
df_order_for_merge = df_order.drop(columns = ['amount', 'discount', 'net_order_amount', 'profit'])

In [28]:
df_order_for_merge

order_date  order_id  customer_id  poc_id  order_status  warehouse_id  \
0     2025-04-03    136349         5235       6             0             3   
1     2025-01-30    110393         7622    7814             0             6   
2     2025-02-12    116427         3827      51             0             3   
3     2024-12-26     96182         2223      25             0             3   
4     2024-12-30     97858         6903    7039             0             1   
...          ...       ...          ...     ...           ...           ...   
64454 2024-12-26     96152          948      17             0             1   
64455 2024-12-26     96153         5822    5602             0             2   
64456 2024-12-25     96166         1655      37             0             3   
64457 2024-12-26     96173         4268    4293             0             2   
64458 2024-12-26     96175          998      24             0             1   

      fiscal_year  
0           25-26  
1           24-25  
2           24-25  
3           24-25  
4           24-25  
...           ...  
64454       24-25  
64455       24-25  
64456       24-25  
64457       24-25  
64458       24-25  

[64459 rows x 7 columns]

In [29]:
df_associated_order

order_id  order_item_id                    item_name  quantity  \
0         104592         202967                 Anannya Rice         2   
1          74112         148560           Ambey Patent Maida         1   
2         105873         205222                    Arhar Dal        10   
3         112302         216104                  Kezar Maida         5   
4          74308         148902                 Normal Sugar         1   
...          ...            ...                          ...       ...   
115088    139435         262494              SAPNA GOLD AATA         2   
115089    139433         262491                 Hum Tum Rice         5   
115090    139431         262489  Masodha Double Filter Sugar         4   
115091    139430         262488                        Matar        10   
115092    139429         262487              SAPNA GOLD AATA         1   

        invoiced_quantity       mrp  price_per_unit   amount  discount_amount  \
0                       2  1732.000          1534.0  3464.00           396.00   
1                       1  1899.000          1760.0  1899.00           139.00   
2                       0   173.567           127.0  1735.67           465.67   
3                       5  1760.000          1652.0  8800.00           540.00   
4                       1  2190.000          2110.0  2190.00            80.00   
...                   ...       ...             ...      ...              ...   
115088                  0  2010.000          1470.0  4020.00          1080.00   
115089                  0  1390.000          1215.0  6950.00           875.00   
115090                  0  2430.000          2210.0  9720.00           880.00   
115091                  0    54.675            45.0   546.75            96.75   
115092                  0  2010.000          1450.0  2010.00           560.00   

        profit  
0      -208.00  
1       -10.00  
2      -377.00  
3      -140.00  
4        40.00  
...        ...  
115088 -780.00  
115089 -375.00  
115090 -160.00  
115091  -56.75  
115092 -410.00  

[115093 rows x 10 columns]

In [30]:
df_final = result = pd.merge(df_associated_order, df_order_for_merge, on='order_id', how='inner')

In [31]:
df_final

order_id  order_item_id                    item_name  quantity  \
0         104592         202967                 Anannya Rice         2   
1          74112         148560           Ambey Patent Maida         1   
2         105873         205222                    Arhar Dal        10   
3         112302         216104                  Kezar Maida         5   
4          74308         148902                 Normal Sugar         1   
...          ...            ...                          ...       ...   
115088    139435         262494              SAPNA GOLD AATA         2   
115089    139433         262491                 Hum Tum Rice         5   
115090    139431         262489  Masodha Double Filter Sugar         4   
115091    139430         262488                        Matar        10   
115092    139429         262487              SAPNA GOLD AATA         1   

        invoiced_quantity       mrp  price_per_unit   amount  discount_amount  \
0                       2  1732.000          1534.0  3464.00           396.00   
1                       1  1899.000          1760.0  1899.00           139.00   
2                       0   173.567           127.0  1735.67           465.67   
3                       5  1760.000          1652.0  8800.00           540.00   
4                       1  2190.000          2110.0  2190.00            80.00   
...                   ...       ...             ...      ...              ...   
115088                  0  2010.000          1470.0  4020.00          1080.00   
115089                  0  1390.000          1215.0  6950.00           875.00   
115090                  0  2430.000          2210.0  9720.00           880.00   
115091                  0    54.675            45.0   546.75            96.75   
115092                  0  2010.000          1450.0  2010.00           560.00   

        profit order_date  customer_id  poc_id  order_status  warehouse_id  \
0      -208.00 2025-01-16         5980    5381             0             2   
1       -10.00 2024-10-22         5540    4680             0             3   
2      -377.00 2025-01-19         1119    5381             0             2   
3      -140.00 2025-02-04         3776      43             0             1   
4        40.00 2024-10-22         5569    5602             0             2   
...        ...        ...          ...     ...           ...           ...   
115088 -780.00 2025-04-10         7869    7228             1             4   
115089 -375.00 2025-04-10         7806    7228             1             4   
115090 -160.00 2025-04-10         7806    7228             1             4   
115091  -56.75 2025-04-10         4982    7228             1             4   
115092 -410.00 2025-04-10         7056    7228             1             4   

       fiscal_year  
0            24-25  
1            24-25  
2            24-25  
3            24-25  
4            24-25  
...            ...  
115088       25-26  
115089       25-26  
115090       25-26  
115091       25-26  
115092       25-26  

[115093 rows x 16 columns]

In [32]:
df_final.to_csv('merge_final.csv', index=False)

In [33]:
from ydata_profiling import ProfileReport  # If using latest version of pandas-profiling

# Generate profile reports
profile_final = ProfileReport(df_final, title="Orders Data Report", explorative=True)
# Save as HTML
profile_final.to_file("df_final_visualization.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]


100%|██████████| 16/16 [00:01<00:00, 15.69it/s]


Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [34]:
print('The total number of unique customer: ',(df_final.customer_id.nunique()))

The total number of unique customer:  4161


In [35]:
df_final.nunique()

order_id              64459
order_item_id        115093
item_name               607
quantity                117
invoiced_quantity       128
mrp                    1992
price_per_unit         2216
amount                 8264
discount_amount        6429
profit                 4614
order_date              190
customer_id            4161
poc_id                   97
order_status              2
warehouse_id              6
fiscal_year               2
dtype: int64

In [36]:
df_final['sales_amount'] = df_final['amount']	- df_final['discount_amount']

In [37]:
df_final.sales_amount.sum()

948639674.0799999

In [44]:
df_final.item_name.value_counts()

item_name
Ruchi Gold Palm Pouch(1L)    6945
Normal Sugar                 6233
Prakash Maida                4952
Prakash Aata                 4779
Shreshtha Aata               4739
                             ... 
RAAGI                           1
Haldi Pkt(1kg)                  1
Kishu Mini Mogra Rice           1
Gari Burada                     1
Agro Chana Besan Mahin          1
Name: count, Length: 607, dtype: int64

In [45]:
df_final.item_name.unique()

array(['Anannya Rice', 'Ambey Patent Maida', 'Arhar Dal', 'Kezar Maida',
       'Normal Sugar', 'Prakash Aata', 'Prakash Maida', 'Matar',
       'DCM Double Filter Sugar', 'Tata Salt Pkt(1Kg)',
       'Adani Patla Besan', 'Ashok Kali Mirch Powder (50 Gm)',
       'Kezar Aata', 'SODA CHINI (1KG) सोडा', 'Soyum Soya Pouch',
       'Sunny Maida', 'Rice MV Gold', 'Shreshtha Aata',
       'Nourish Sunflower Pouch', 'Bail Kolhu Pouch', 'Arrarot',
       'Swastik maida', 'Fortune Soya Pouch(1L)', 'Steam Rice',
       'Swastik Suji Rava', 'Ruchi Gold Palm Pouch(1L)', 'Medium Chhola',
       'Laal Mirch Sabut', 'Ganesh Maida', 'Shri Ram maida (50kg)',
       'Rajdhani Chakki aata', 'Rajdhani Maida', 'Pani (500 Ml)',
       'Pukhraj Arhar Dal', 'Parivar Pasand Chakki Aata (50kg)',
       'Soyum Soya (15Lit)', 'Laal Gulaab Maida', 'SAPNA GOLD MAIDA',
       'Chana Dal', 'Adani Mota Besan', 'Fryola Palm', 'Kaali Masoor Dal',
       'Aashirwad Chakki Atta (10kg)', 'Safal Atta',
       'Nature Fresh 

In [56]:
((df_final.mrp - df_final.price_per_unit) * 100 / df_final.price_per_unit).mean()

10.57309479672752